### PR TITLE
Implement Rent, Return and Condition rendering by role

### PR DESCRIPTION
**Pull request by Kimara1904**
Implement Rent on Book Detail and Book Card
Implement Return and Rent History on Book Detail
Getting role from token and by it render some components

### CODE REVIEW

File: fe-library-app/src/AppContent.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const AppContent = () => {
       <div className='App-sideBar'>
         <NavigationBar onNavClick={onNavClickHandler} />
         <SideBar
-          display={isThereToken()}
+          display={isThereToken() && (isAdmin() || isLibrarian())}
```
Comment (sttevan) [inline fe-library-app/src/AppContent.tsx:32]: Necemo ovo da zovemo isThereToken nego isUserLoggedIn

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 2]
```diff
@@ -2,16 +2,21 @@ import { useCallback, useEffect, useState } from 'react'
...
 import DefaultBookCover from '../BookList/BookCard/DefaultBookCover.png'
 import style from './BookDetail.module.css'
 
 const BookDetail = () => {
   const { id } = useParams()
   const [ book, setBook ] = useState<BookByIdItemResponse>()
+  const [ rentalHistory, setRentalHistory ] = useState<RentHistoryResponse[]>([])
   const navigate = useNavigate()
   const [ showUpdateBookModal, setShowUpdateBookModal ] = useState(false)
+  const isAuthorized = isAdmin() || isLibrarian()
```
Comment (sttevan) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:19]: ne moze ovo da se zove isAuthorized
smisli bolje ime

File: fe-library-app/src/components/BookDetail/BookDetail.tsx [hunk 3]
```diff
@@ -44,6 +63,48 @@ const BookDetail = () => {
...
+          }
+        })
+      })
+      .catch(() => {
+        alert('Error with returning book')
+      })
+  }
+
+  const handleRentBook = () => {
+    if ((book?.Available as number) > 0) {
```
Comment (sttevan) [inline fe-library-app/src/components/BookDetail/BookDetail.tsx:86]: Necemo ovu proveru ovde da radimo, nego cemo disablovati rent dugme ako nemamo available books,

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx [hunk 4]
```diff
@@ -16,15 +19,19 @@ const BookCard = (props: BookProp) => {
   const navigator = useNavigate()
   const [ showUpdateModal, setShowUpdateModal ] = useState(false)
 
-  const handleModifyClick = () => {
+  const isAuthorized = isAdmin() || isLibrarian()
```
Comment (sttevan) [inline fe-library-app/src/components/BookList/BookCard/BookCard.tsx:22]: evo ga opet

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx [hunk 5]
```diff
@@ -37,14 +44,29 @@ const BookCard = (props: BookProp) => {
     }
   }
 
+  const handleRentBook = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.stopPropagation()
+    if (props.book.Available> 0) {
```
Comment (sttevan) [inline fe-library-app/src/components/BookList/BookCard/BookCard.tsx:49]: opet ova provera

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx [hunk 6]
```diff
@@ -37,14 +44,29 @@ const BookCard = (props: BookProp) => {
...
+          alert('Error with renting book')
+        })
+    } else {
+      alert('Book is not available for renting')
+    }
+  }
+
   return (
-    <div className={styles.book_card}>
+    <div className={styles.book_card} onClick={handleImageClick}>
```
Comment (sttevan) [inline fe-library-app/src/components/BookList/BookCard/BookCard.tsx:64]: Ne klikcemo na image nego na bookCard

File: fe-library-app/src/components/NavigationBar/NavigationBar.tsx [hunk 7]
```diff
@@ -26,7 +27,7 @@ const NavigationBar = (props: NavigationBarProps) => {
             <Link to='/profile'>Profile</Link>
           </li>
         )}
-        {loggedIn && (
+        {(loggedIn && (isAdmin() || isLibrarian())) && (
```
Comment (sttevan) [inline fe-library-app/src/components/NavigationBar/NavigationBar.tsx:30]: samo preimenuj metodu i pozovi je direktno

**Review by sttevan — CHANGES_REQUESTED** [review]

File: fe-library-app/src/AppContent.tsx (hunk 1 above)
Comment (Kimara1904) [reply fe-library-app/src/AppContent.tsx:32]: Jasno. Uradjeno

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx (hunk 5 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookList/BookCard/BookCard.tsx:49]: Reseno

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx (hunk 6 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookList/BookCard/BookCard.tsx:64]: Prepravljeno

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx [hunk 8]
```diff
@@ -65,19 +87,34 @@ const BookCard = (props: BookProp) => {
...
       </div>
       <div className={styles.extra_func}>
-        <button className={styles.book_phone_button} onClick={handleModifyClick}>
-          Modify
-        </button>
-        <button className={styles.book_desktop_button} onClick={() => setShowUpdateModal(true)}>
-          Modify
-        </button>
-        <button onClick={handleDeleteClick}>Delete</button>
+        {isThereToken() && <button onClick={handleRentBook}>Rent</button>}
```
[the remark this answers is not in the document]
Comment (Kimara1904) [reply fe-library-app/src/components/BookList/BookCard/BookCard.tsx:90]: Reseno

File: fe-library-app/src/components/BookDetail/BookDetail.tsx (hunk 2 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookDetail/BookDetail.tsx:19]: isAdminOrLibrarian mozda?

Comment (sttevan) [reply fe-library-app/src/components/BookDetail/BookDetail.tsx:19]: moze

File: fe-library-app/src/components/BookList/BookCard/BookCard.tsx (hunk 4 above)
Comment (Kimara1904) [reply fe-library-app/src/components/BookList/BookCard/BookCard.tsx:22]: Reseno

File: fe-library-app/src/components/NavigationBar/NavigationBar.tsx (hunk 7 above)
Comment (Kimara1904) [reply fe-library-app/src/components/NavigationBar/NavigationBar.tsx:30]: celu proveru ili samo ovaj deo za Admina i Librariana?

Comment (sttevan) [reply fe-library-app/src/components/NavigationBar/NavigationBar.tsx:30]: mislim na loggedin

Comment (Kimara1904) [reply fe-library-app/src/components/NavigationBar/NavigationBar.tsx:30]: Aha jasno reseno

**Review by vladimir-agbaba-valcon — APPROVED** [review]

**Pull request merged**